lib/wibox/widget/template.lua

```diff
@@ -5,10 +5,27 @@
 -- concrete widget definition. The template widget can be used to build widgets
 -- that the user can customize at their will, thanks to the template mechanism.
 --
+-- Common usage examples
+-- =====================
+--
+-- A basic implementation of the template widget needs a widget definition and
+-- a callback to manage widget updates.
+--
+-- The `:update()` can be called later to update the widget template. Arguments
+-- can be provided to the `:update()` method, it will be forwarded to the
+-- `update_callback` function.
```

Comment:
I think this should be moved to a separate section, e.g. "how to create a widget/an API that utilizes widget templates".

The common usage that most users will run into will only be about things like "what to write in `awful.taglist { widget_template = ... }`?". For those cases, the user will likely just get confused about where they are supposed to call `:update()` and with what arguments.

In the examples, the same distinction should be made clear, with comments like
```lua
-- this is what the user/consumer will write in their `rc.lua`
```
```lua
-- this is what a widget creator will use in their widget's code, e.g. `lib/awful/taglist.lua`
```

As it is right now, users will likely assume that _they_ are supposed to call `:update()` in their `rc.lua`.

Reply:
I agree this can be confusing since with this widget, you have 2 use case : as the widget author and as the end-user. I'll try to improve this documentation part.

Thank you for your input! :smile: 

(As a quick note : the taglist/tasklist widget are not targets for this widget template implementation. While it is possible to use it as a base for these list widgets, I think we should have a separate list_template widget to handle all the layouting and iterate-over-a-list stuff that these two need.)